test/hotspot/jtreg/vmTestbase/nsk/share/jdi/JDIBase.java

```
@@ -119,11 +119,14 @@ protected final BreakpointRequest settingBreakpoint(ThreadReference thread,
                 try {
                     breakpRequest = eventRManager.createBreakpointRequest(lineLocation);
                     breakpRequest.putProperty("number", property);
-                    breakpRequest.addThreadFilter(thread);
+                    if (thread != null) {
```

The change might hide failure if thread is not set in the test.
I would prefer to have
private settingBreakpoint(ThreadReference thread,.,.) that allows null
and 
```
protected final BreakpointRequest settingBreakpoint(ThreadReference thread,
                                                     ReferenceType testedClass,
                                                     String methodName,
                                                     String bpLine,
                                                     String property)
```
that still fails early if thread is null. (I think now it should fail in 
`
 breakpRequest.addThreadFilter(thread);
`
string. 
So for any test that don't have proper thread - test fails early when setting breakpoint and not because it hasn't find it.

---

Ok. I updated to use a different API that implies no thread.